person.rb

```diff
@@ -1,18 +1,32 @@
-class Person
+require_relative 'nameable'
+require_relative 'rentals'
+
+class Person < Nameable
   attr_reader :id
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
     @id = rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+    rental.book.add_rental(rental)
+  end
```

Comment:
- [ ] I think this way of creating this method will result in duplicate rentals for a person object. In the `add_rental` method in the Book class, you are adding `rental` again to the Person object after adding it here. I would strongly recommend you modify this method to simply instantiate a rental object given a book and date and then modify the Rental constructor to add `rental` to the respective arrays.

  ```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
  ```

Reply:
Fixed! 🔨